x-pack/plugins/security_solution/public/resolver/store/data/action.ts

```diff
@@ -111,7 +116,7 @@ interface ServerFailedToReturnCurrentRelatedEventData {
 
 interface ServerReturnedCurrentRelatedEventData {
   readonly type: 'serverReturnedCurrentRelatedEventData';
-  readonly payload: SafeResolverEvent;
+  readonly payload: SafeResolverEvent & { dataRequestID?: number };
```

Comment:
Why add `dataRequestID` to the event? That might replace an existing key with the same name. How about this instead:

```
readonly payload: { event: SafeResolverEvent; dataRequestID: number }
```